### PR TITLE
- refactor AIO DR

### PR DESCRIPTION
**Pull request by erz4**

### CODE REVIEW

File: aws/PAS-AIO-DR-template.yaml [hunk 1]
```diff
@@ -0,0 +1,1227 @@
...
+    Properties: 
+      LogGroupName: !Sub '${LogGroup}'
+      LogStreamName: 'VaultMachine/VaultInitLog'
+    DeletionPolicy: Retain
+  VaultPostInstallLogStream:
+    Type: AWS::Logs::LogStream
+    Properties: 
+      LogGroupName: !Sub '${LogGroup}'
+      LogStreamName: 'VaultMachine/VaultPostInstallLog'
+  VaultCfnInitLogStream:
```
Comment (yogevh) [inline aws/PAS-AIO-DR-template.yaml]: duplicated resource
  VaultCfnInitLogStream:

File: aws/PAS-AIO-DR-template.yaml [hunk 2]
```diff
@@ -0,0 +1,1227 @@
...
+      LogGroupName: !Sub '${LogGroup}'
+      LogStreamName: 'VaultMachine/CfnInitLog'
+    DeletionPolicy: Retain
+  VaultInitLogStream:
+    Type: AWS::Logs::LogStream
+    Properties: 
+      LogGroupName: !Sub '${LogGroup}'
+      LogStreamName: 'VaultMachine/VaultInitLog'
+    DeletionPolicy: Retain
+  VaultPostInstallLogStream:
```
Comment (yogevh) [inline aws/PAS-AIO-DR-template.yaml]: i didn't see this log in Full-PAS or in AIO or SingleVault, do we need it there as well?

File: aws/PAS-AIO-DR-template.yaml [hunk 3]
```diff
@@ -0,0 +1,1227 @@
...
+          <script>
+          cfn-init.exe -v -s ${AWS::StackId} -r VaultMachine --configsets ascending --region ${AWS::Region}
+          </script>
+      KeyName: !Ref 'KeyName'
+      IamInstanceProfile: !Ref 'VaultInstancesProfile'
+    Metadata:
+      AWS::CloudFormation::Init:
+        configSets:
+          ascending:
+            - configSSMAndHostname
```
Comment (yogevh) [inline aws/PAS-AIO-DR-template.yaml]: please change the name of configset according to AIO deployment
we don't change the hostname so we can remove HostName from the name

File: aws/PAS-AIO-DR-template.yaml [hunk 4]
```diff
@@ -0,0 +1,1227 @@
...
+            - configServices
+            - configSignal
+        configSSMAndHostname:
```
Comment (yogevh) [inline aws/PAS-AIO-DR-template.yaml]: remove Hostname

File: aws/PAS-AIO-DR-template.yaml [hunk 5]
```diff
@@ -0,0 +1,1227 @@
...
+        'Fn::Sub': >-
+          <script> cfn-init.exe -v -s ${AWS::StackId} -r VaultDRMachine
+          --configsets ascending --region ${AWS::Region} </script>
+      KeyName: !Ref KeyName
+      IamInstanceProfile: !Ref VaultInstancesProfile
+    Metadata:
+      'AWS::CloudFormation::Init':
+        configSets:
+          ascending:
+            - configSSMAndHostname
```
Comment (yogevh) [inline aws/PAS-AIO-DR-template.yaml]: same here, since we don't change the hostname we should alter the name

File: aws/PAS-AIO-DR-template.yaml [hunk 6]
```diff
@@ -0,0 +1,1227 @@
...
+        configSignal:
+          commands:
+            0-signalCompletion:
+              command: !Sub >
+                cfn-signal.exe -e %ERRORLEVEL% --stack ${AWS::StackId} --resource VaultMachine --region ${AWS::Region}
+              waitAfterCompletion: 0
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT15M
+  VaultDRMachine:
```
Comment (yogevh) [inline aws/PAS-AIO-DR-template.yaml]: why do we have 2 VaultDRMachine :)

File: aws/PAS-AIO-DR-template.yaml [hunk 7]
```diff
@@ -0,0 +1,1227 @@
...
+              command: !Sub >
+                  powershell.exe -File C:\CyberArk\Deployment\PSMRegistration.ps1
+                  -VaultAdminUser "Administrator" -SSMAdminPassParameterID ${StoreAdminPassword.SsmId}
+              waitAfterCompletion: '0'  
+            304-PSMserviceConfig:
+              command: sc config "Cyber-Ark Privileged Session Manager" start=auto
+              waitAfterCompletion: '0'
+        configSignal:  
+          commands:
+            98-SignalCompletion:
```
Comment (yogevh) [inline aws/PAS-AIO-DR-template.yaml]: first restart the instance, and then signal for completion
change the order of commands

File: aws/PAS-AIO-DR-template.yaml [hunk 8]
```diff
@@ -0,0 +1,1227 @@
...
+      KeyName: !Ref KeyName
+      IamInstanceProfile: !Ref VaultInstancesProfile
+    Metadata:
+      'AWS::CloudFormation::Init':
+        configSets:
+          ascending:
+            - configSSMAndHostname
+            - configServices
+            - configSignal
+        configSSMAndHostname:
```
Comment (yogevh) [inline aws/PAS-AIO-DR-template.yaml]: remove hostname

File: aws/PAS-AIO-DR-template.yaml [hunk 9]
```diff
@@ -0,0 +1,1227 @@
...
+              except Exception as E:
+                  print (E)
+                  return cfnresponse.send(event, context, cfnresponse.FAILED, {}, physicalResourceId)
+
+      Runtime: python3.7
+      Handler: index.lambda_handler
+      Timeout: 60
+      Role: !GetAtt 'LambdaDeployRole.Arn'
+    Condition: EULACondition
+  CopyfileFromBucketLambda:
```
Comment (yogevh) [inline aws/PAS-AIO-DR-template.yaml]: remove irrlevant lambda

File: aws/PAS-AIO-DR-template.yaml (hunk 3 above)
Comment (erz4) [reply aws/PAS-AIO-DR-template.yaml]: changed

File: aws/PAS-AIO-DR-template.yaml (hunk 4 above)
Comment (erz4) [reply aws/PAS-AIO-DR-template.yaml]: changed

File: aws/PAS-AIO-DR-template.yaml (hunk 5 above)
Comment (erz4) [reply aws/PAS-AIO-DR-template.yaml]: changed

File: aws/PAS-AIO-DR-template.yaml (hunk 8 above)
Comment (erz4) [reply aws/PAS-AIO-DR-template.yaml]: changed

File: aws/PAS-AIO-DR-template.yaml (hunk 1 above)
Comment (erz4) [reply aws/PAS-AIO-DR-template.yaml]: deleted

File: aws/PAS-AIO-DR-template.yaml (hunk 6 above)
Comment (erz4) [reply aws/PAS-AIO-DR-template.yaml]: ctrl-V instead of ctrl-X

File: aws/PAS-AIO-DR-template.yaml (hunk 7 above)
Comment (erz4) [reply aws/PAS-AIO-DR-template.yaml]: changed the order

File: aws/PAS-AIO-DR-template.yaml (hunk 9 above)
Comment (erz4) [reply aws/PAS-AIO-DR-template.yaml]: deleted

File: aws/PAS-AIO-DR-template.yaml (hunk 2 above)
Comment (erz4) [reply aws/PAS-AIO-DR-template.yaml]: exist in AIO

Comment (yogevh) [reply aws/PAS-AIO-DR-template.yaml]: can you please check with Liora about Full-PAS? and with Avishay about Single-Vault?

**Pull request merged**